### PR TITLE
Add better code docs to storage refactor

### PR DESCRIPTION
**Pull request by zhilingc**
**What this PR does / why we need it**:
Reopening this after rebasing on master.
Addresses most of @woop 's comments on PR #567

### CODE REVIEW

**Comment by woop** [comment]
Let me know if you want this reviewed or not.

**Comment by zhilingc** [comment]
> Let me know if you want this reviewed or not.

Yeah, I had to close the other one because of some issues rebasing. I added changes based on your comment already

File: serving/src/main/java/feast/serving/configuration/ServingServiceConfig.java [hunk 1]
```diff
@@ -117,7 +117,8 @@ public ServingService servingService(
                 .setStorage(storage)
                 .build();
 
-        servingService = new HistoricalServingService(bqRetriever, specService, jobService);
+        servingService =
+            new HistoricalServingService(bqRetriever, specService, jobService, jobStagingLocation);
```
Comment (woop) [inline serving/src/main/java/feast/serving/configuration/ServingServiceConfig.java:121]: Trust me I really want to greenlight this PR, but I don't think adding `stagingLocation` directly to the `HistoricalServingService` is correct here. That state is already stored within the retriever, which means it is also available to the HistoricalServingService if we expose it.

Comment (woop) [reply serving/src/main/java/feast/serving/configuration/ServingServiceConfig.java:121]: If you are happy with me making the change then I can quickly do it and you can see if you agree with the structure?

**Pull request merged**